### PR TITLE
Add estimate tx dependencies to docs

### LINKED ISSUE #754: Add section in the docs for transaction dependencies estimation
We now support tx dependencies estimation, but looks like the docs don't mention it. We should add a section for it.

### PR DESCRIPTION
**Pull request by MujkicA**
Closes #754

- slightly adjusted the section on output variables and extended it with message outputs
- added section on dependency estimation

### CODE REVIEW

File: docs/src/calling-contracts/tx-dependency-estimation.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+```rust,ignore
+{{#include ../../../examples/contracts/src/lib.rs:dependency_estimation}}
+```
+
+The minimal number of attempts corresponds to the number of external contracts and output variables needed and defaults to 10.
+
+> **Note:** `.estimate_tx_dependencies()` can also be used when working with multi calls.
+
+> **Note:** if no resolution was found after exhausting all simulation attempts, the last received error will be propagated. The same will happen if an error occurs that is not related to the transactions dependencies.
```
Comment (digorithm) [inline docs/src/calling-contracts/tx-dependency-estimation.md:27]: Improving clarity and conciseness of the last sentence:

```suggestion
> **Note:** if no resolution was found after exhausting all simulation attempts, the last received error will be propagated. The same will happen if an error is unrelated to transaction dependencies.
```

File: docs/src/calling-contracts/variable-outputs.md [hunk 2]
```diff
@@ -1,17 +1,29 @@
-# Variable outputs
+# Output variables
```
Comment (digorithm) [inline docs/src/calling-contracts/variable-outputs.md:1]: Shouldn't it be based on `VariableOutput`, i.e., keep it as `Variable outputs` (considering the client to be the source of truth here https://github.com/FuelLabs/fuel-core/blob/a3f584f12bb3e160aa00e08616a27c81734ebd02/crates/fuel-core/src/schema/tx/output.rs#L84)?

File: docs/src/calling-contracts/variable-outputs.md [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
 
-In some cases, you might want to send funds to the output of a transaction. Sway has a specific method for that: `transfer_to_address`(coins, asset_id, recipient)`. So, if you have a contract that does something like this:
+In some cases, the contract you are calling might be transferring funds to a specific address, depending on its execution. The underlying transaction for such a contract call has to have the appropriate number of [variable outputs](https://github.com/FuelLabs/fuel-specs/blob/master/src/protocol/tx_format/output.md#outputvariable) to succeed.
```
Comment (digorithm) [inline docs/src/calling-contracts/variable-outputs.md:3]: Some conciseness nits:

```suggestion
Sometimes, the contract you call might transfer funds to a specific address, depending on its execution. The underlying transaction for such a contract call has to have the appropriate number of [variable outputs](https://github.com/FuelLabs/fuel-specs/blob/master/src/protocol/tx_format/output.md#outputvariable) to succeed.
```

File: docs/src/calling-contracts/variable-outputs.md [hunk 4]
```diff
@@ -1,17 +1,29 @@
...
+
+Let's say you deployed a contract with the following method:
 
 ```rust,ignore
 {{#include ../../../packages/fuels/tests/contracts/token_ops/src/main.sw:variable_outputs}}
 ```
 
-With the SDK, you can call `transfer_coins_to_output` by chaining `append_variable_outputs(amount)` to your contract call. Like this:
+When calling `transfer_coins_to_output` with the SDK, you can specify the number of output variables by chaining `append_variable_outputs(amount)` to your call. Like this:
```
Comment (digorithm) [inline docs/src/calling-contracts/variable-outputs.md:11]:
```suggestion
When calling `transfer_coins_to_output` with the SDK, you can specify the number of variable outputs by chaining `append_variable_outputs(amount)` to your call. Like this:
```

**Review by digorithm — CHANGES_REQUESTED** [review]
Looking really good! Thanks for doing this doc work.

File: docs/src/calling-contracts/variable-outputs.md (hunk 2 above)
Comment (MujkicA) [reply docs/src/calling-contracts/variable-outputs.md:1]: Yup, I got confused

**Review by iqdecay — APPROVED** [review]
This is GREAT! Actually cleared out some confusion I had myself <3

**Review by hal3e — APPROVED** [review]

**Review by Salka1988 — APPROVED** [review]

**Review by digorithm — APPROVED** [review]
Wonderful 😁

**Pull request merged**